tendrl/commons/objects/cluster/atoms/clear_cluster_details/__init__.py

```diff
@@ -0,0 +1,120 @@
+from tendrl.commons.objects.cluster_tendrl_context \
+    import ClusterTendrlContext
+from tendrl.commons.objects.node import Node
+
+
+class ClearClusterDetails(objects.BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(ClearClusterDetails, self).__init__(*args, **kwargs)
+
+    def run(self):
+        integration_id = self.parameters['TendrlContext.integration_id']
+        cluster_tendrl_context = ClusterTendrlContext(
+            integration_id=integration_id
+        ).load()
+
+        try:
+            NS._int.client.delete(
+                "/clusters/%s" % integration_id,
+                recursive=True
+            )
+            NS._int.client.delete(
+                "/alerting/clusters/%s" % integration_id,
+                recursive=True
+            )
+            NS._int.client.delete(
+                "/indexes/tags/detected_cluster_id_to_integration_id/%s" %
+                cluster_tendrl_context.cluster_id
+            )
+            NS._int.client.delete(
+                "/indexes/tags/provisioner/%s" %
+                integration_id
+            )
+            NS._init.client.delete(
+                "/indexes/tags/tendrl/integration/%s" %
+                integration_id
+            )
+
+            nodes = NS._init.client.read(
+                "/clusters/%s/nodes" % integration_id
+            )
+            node_ids = []
+            for node_id in nodes.leaves:
+                node_ids.append(node_id)
+                nc = Node(node_id=node_id).load()
+                node_ips[node_id] = nc.fqdn
+                NS._int.client.delete(
+                    "/nodes/%s" % node_id,
+                    recursive=True
+                )
+                NS._int.client.delete(
+                    "/alerting/nodes/%s" % node_id,
+                    recursive=True
+                )
+                NS._init.client.delete(
+                    "/indexes/tags/tendrl/node_%s" % node_id
+                )
+
+            # Remove the nodes from /indexes/ip
+            idx_ips = NS._int.client.read("/indexes/ip")
+            for node_id_for_ip in idx_ips.leaves:
+                if node_id_for_ip.value in node_ids:
+                    NS._int.client.delete(
+                        "/indexes/ip/%s" % node_id_for_ip.key
+                    )
+
+            # Remove the network details for cluster nodes
+            nws = NS._int.client.read("/networks")
+            for nw in nws.leaves:
+                nodes_nw = NS._int.client.read("/networks/%s" % nw)
+                for node_nw in nodes_nw.leaves:
+                    if node_nw in node_ids:
+                        NS._int.client.delete(
+                            "/networks/%s/%s" % (nw, node_nw),
+                            recursive=True
+                        )
+
+            # Load the gluster servers list and remove the cluster nodes
+            gl_srvr_list = NS._int.client.read(
+                "/indexes/tags/gluster/server"
```

Comment:
This will also delete any gluster nodes from some other cluster in a multi cluster env. Please remove this

Reply:
We are getting the list and remove only the nodes of the cluster from the list and then update. Below lines take care of the same.